From 9c9e0c29bd5864f93644e038cc736c340f729357 Mon Sep 17 00:00:00 2001
From: "Subhransu S. Prusty" <subhransu.sekhar.prusty@xilinx.com>
Date: Thu, 2 Jan 2020 11:26:49 +0530
Subject: [PATCH] Add transform_meta in v4l2transform plugin

Signed-off-by: Subhransu S. Prusty <subhransu.sekhar.prusty@xilinx.com>
---
 sys/v4l2/gstv4l2transform.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/sys/v4l2/gstv4l2transform.c b/sys/v4l2/gstv4l2transform.c
index edd6c428f..a32066011 100644
--- a/sys/v4l2/gstv4l2transform.c
+++ b/sys/v4l2/gstv4l2transform.c
@@ -997,6 +997,13 @@ gst_v4l2_transform_transform (GstBaseTransform * trans, GstBuffer * inbuf,
   return GST_FLOW_OK;
 }
 
+static gboolean
+gst_v4l2_transform_meta (GstBaseTransform * trans, GstBuffer * outbuf,
+    GstMeta * meta, GstBuffer * inbuf)
+{
+  return TRUE;
+}
+
 static gboolean
 gst_v4l2_transform_sink_event (GstBaseTransform * trans, GstEvent * event)
 {
@@ -1167,6 +1174,9 @@ gst_v4l2_transform_class_init (GstV4l2TransformClass * klass)
 
   base_transform_class->passthrough_on_same_caps = TRUE;
 
+  base_transform_class->transform_meta =
+      GST_DEBUG_FUNCPTR (gst_v4l2_transform_meta);
+
   element_class->change_state =
       GST_DEBUG_FUNCPTR (gst_v4l2_transform_change_state);
 
