diff --git a/drivers/media/platform/xilinx/xilinx-demosaic.c b/drivers/media/platform/xilinx/xilinx-demosaic.c
index 531e6ec..700eb0f 100644
--- a/drivers/media/platform/xilinx/xilinx-demosaic.c
+++ b/drivers/media/platform/xilinx/xilinx-demosaic.c
@@ -59,7 +59,7 @@ struct xdmsc_dev {
 	u32 max_width;
 	u32 max_height;
 };
-
+#if 0
 static inline u32 xdmsc_read(struct xdmsc_dev *xdmsc, u32 reg)
 {
 	u32 data;
@@ -81,6 +81,7 @@ static inline void xdmsc_write(struct xdmsc_dev *xdmsc, u32 reg, u32 data)
 			"Wrote 0x%x does not match read back", data);
 #endif
 }
+#endif
 
 static inline struct xdmsc_dev *to_xdmsc(struct v4l2_subdev *subdev)
 {
@@ -107,6 +108,7 @@ static int xdmsc_s_stream(struct v4l2_subdev *subdev, int enable)
 {
 	struct xdmsc_dev *xdmsc = to_xdmsc(subdev);
 
+#if 0
 	if (!enable) {
 		dev_dbg(xdmsc->xvip.dev, "%s : Off", __func__);
 		gpiod_set_value_cansleep(xdmsc->rst_gpio,
@@ -124,6 +126,7 @@ static int xdmsc_s_stream(struct v4l2_subdev *subdev, int enable)
 
 	/* Start Demosaic Video IP */
 	xdmsc_write(xdmsc, XDEMOSAIC_AP_CTRL, XDEMOSAIC_STREAM_ON);
+#endif
 	return 0;
 }
 
@@ -180,7 +183,7 @@ static int xdmsc_set_format(struct v4l2_subdev *subdev,
 				   XDEMOSAIC_MIN_HEIGHT, xdmsc->max_height);
 
 	if (fmt->pad == XVIP_PAD_SOURCE) {
-		if (__format->code != MEDIA_BUS_FMT_RBG888_1X24) {
+		if (__format->code != MEDIA_BUS_FMT_UYVY8_1X16) {
 			dev_dbg(xdmsc->xvip.dev,
 				"%s : Unsupported source media bus code format",
 				__func__);
@@ -238,7 +241,7 @@ static const struct v4l2_subdev_ops xdmsc_ops = {
 static const struct media_entity_operations xdmsc_media_ops = {
 	.link_validate = v4l2_subdev_link_validate,
 };
-
+#if 1
 static int xdmsc_parse_of(struct xdmsc_dev *xdmsc)
 {
 	struct device *dev = xdmsc->xvip.dev;
@@ -295,7 +298,7 @@ static int xdmsc_parse_of(struct xdmsc_dev *xdmsc)
 	}
 	return 0;
 }
-
+#endif
 static int xdmsc_probe(struct platform_device *pdev)
 {
 	struct xdmsc_dev *xdmsc;
@@ -310,12 +313,13 @@ static int xdmsc_probe(struct platform_device *pdev)
 	rval = xdmsc_parse_of(xdmsc);
 	if (rval < 0)
 		return rval;
+#if 0
 	rval = xvip_init_resources(&xdmsc->xvip);
 
 	/* Reset Demosaic IP */
 	gpiod_set_value_cansleep(xdmsc->rst_gpio,
 				 XDEMOSAIC_RESET_DEASSERT);
-
+#endif
 	/* Init V4L2 subdev */
 	subdev = &xdmsc->xvip.subdev;
 	v4l2_subdev_init(subdev, &xdmsc_ops);
@@ -342,7 +346,7 @@ static int xdmsc_probe(struct platform_device *pdev)
 	*def_fmt = xdmsc->default_formats[XVIP_PAD_SINK];
 
 	/* Source Pad has a fixed media bus format of RGB */
-	def_fmt->code = MEDIA_BUS_FMT_RBG888_1X24;
+	def_fmt->code = MEDIA_BUS_FMT_UYVY8_1X16;
 	xdmsc->formats[XVIP_PAD_SOURCE] = *def_fmt;
 
 	xdmsc->pads[XVIP_PAD_SINK].flags = MEDIA_PAD_FL_SINK;
@@ -367,7 +371,7 @@ static int xdmsc_probe(struct platform_device *pdev)
 v4l2_subdev_error:
 	media_entity_cleanup(&subdev->entity);
 media_error:
-	xvip_cleanup_resources(&xdmsc->xvip);
+	//xvip_cleanup_resources(&xdmsc->xvip);
 	return rval;
 }
 
@@ -378,7 +382,7 @@ static int xdmsc_remove(struct platform_device *pdev)
 
 	v4l2_async_unregister_subdev(subdev);
 	media_entity_cleanup(&subdev->entity);
-	xvip_cleanup_resources(&xdmsc->xvip);
+	//xvip_cleanup_resources(&xdmsc->xvip);
 	return 0;
 }
 
